core/templates/components/common-layout-directives/navigation-bars/top-navigation-bar.component.ts

```diff
@@ -0,0 +1,381 @@
+      this.navigationService.KEYBOARD_EVENT_TO_KEY_CODES;
+    this.windowIsNarrow = this.windowDimensionsService.isWindowNarrow();
+
+    let service = this.classroomBackendApiService;
+    service.fetchClassroomPromosAreEnabledStatusAsync().then(
+      (classroomPromosAreEnabled) => {
+        this.CLASSROOM_PROMOS_ARE_ENABLED = classroomPromosAreEnabled;
+      });
+
+    this.directiveSubscriptions.add(
+      this.searchService.onSearchBarLoaded.subscribe(
+        () => {
+          setTimeout(this.truncateNavbar, 100);
+        }
+      )
+    );
+
+    this.userService.getUserInfoAsync().then((userInfo) => {
+      if (userInfo.getPreferredSiteLanguageCode()) {
+        // This.translate.use(userInfo.getPreferredSiteLanguageCode());
+        this.i18nLanguageCodeService.setI18nLanguageCode(
+          userInfo.getPreferredSiteLanguageCode());
+      }
+      this.isModerator = userInfo.isModerator();
+      this.isAdmin = userInfo.isAdmin();
+      this.isTopicManager = userInfo.isTopicManager();
+      this.isSuperAdmin = userInfo.isSuperAdmin();
+      this.userIsLoggedIn = userInfo.isLoggedIn();
+      this.username = userInfo.getUsername();
+      if (this.username) {
+        this.profilePageUrl = this.urlInterpolationService.interpolateUrl(
+          '/profile/<username>', {
+            username: this.username
+          });
+      }
+
+      if (this.userIsLoggedIn) {
+        this.userBackendApiService.showUnseenNotifications();
+      }
+    });
+
+    for (var i = 0; i < this.NAV_ELEMENTS_ORDER.length; i++) {
+      this.navElementsVisibilityStatus[this.NAV_ELEMENTS_ORDER[i]] = true;
+    }
+
+    this.directiveSubscriptions.add(
+      this.windowDimensionsService.getResizeEvent().subscribe(evt => {
+        this.windowIsNarrow = this.windowDimensionsService.isWindowNarrow();
+        // If window is resized larger, try displaying the hidden
+        // elements.
+        if (
+          this.currentWindowWidth < this.windowDimensionsService.getWidth()) {
+          for (var i = 0; i < this.NAV_ELEMENTS_ORDER.length; i++) {
+            if (
+              !this.navElementsVisibilityStatus[
+                this.NAV_ELEMENTS_ORDER[i]]) {
+              this.navElementsVisibilityStatus[this.NAV_ELEMENTS_ORDER[i]] =
+                true;
+            }
+          }
+        }
+
+        // Close the sidebar, if necessary.
+        this.sidebarStatusService.closeSidebar();
+        this.sidebarIsShown = this.sidebarStatusService.isSidebarShown();
+        this.currentWindowWidth = this.windowDimensionsService.getWidth();
+        this.debouncerService.debounce(this.truncateNavbar, 500);
+      })
+    );
+    // The function needs to be run after i18n. A timeout of 0 appears
+    // to run after i18n in Chrome, but not other browsers. The
+    // will check if i18n is complete and set a new timeout if it is
+    // not. Since a timeout of 0 works for at least one browser,
+    // it is used here.
```

Comment:
Ummm, a bit concered about this.. We need to fix this in this PR...

Reply:
I'm not very sure about why is this used here or what can be it's alternative, but I think it is needed due to single-threaded nature of js, a bit of delay is required so that truncateNavbar is executed just after our i18n loads.Can you please shed some more light on this?

Reply:
Oh was this an existing comment?

Reply:
yes